In [1]:
import pandas as pd
import json
import math

In [2]:
df = pd.read_json("./experiment_queries_execution_boost_rerank2.json", orient="records")
df
    

method_id  user_id  class_id  \
0                          single_elastic_bm25_boost_0.25        1  20825628   
1                          single_elastic_bm25_boost_0.25        1  20825628   
2                          single_elastic_bm25_boost_0.25        1  20825628   
3                          single_elastic_bm25_boost_0.25        1  20825628   
4                          single_elastic_bm25_boost_0.25        1  20825628   
...                                                   ...      ...       ...   
298579  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298580  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298581  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298582  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298583  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   

                       query_id               class_category        rr  \
0                  user_query_0           instances_100_1000  0.000000   
1                  user_query_1           instances_100_1000  0.000000   
2                  user_query_2           instances_100_1000  0.000000   
3                  user_query_3           instances_100_1000  0.000000   
4                  user_query_4           instances_100_1000  0.000000   
...                         ...                          ...       ...   
298579  mech_label_user_query_1  instances_100000_1000000000  1.000000   
298580  mech_label_user_query_2  instances_100000_1000000000  1.000000   
298581  mech_label_user_query_3  instances_100000_1000000000  1.000000   
298582  mech_label_user_query_4  instances_100000_1000000000  0.333333   
298583  mech_label_user_query_5  instances_100000_1000000000  1.000000   

            time  
0       0.110425  
1       0.071166  
2       0.051668  
3       0.048177  
4       0.049556  
...          ...  
298579  0.091356  
298580  0.087534  
298581  0.090200  
298582  0.096017  
298583  0.089052  

[298584 rows x 7 columns]

# Overall

In [3]:
df_thin = df[["method_id", "rr", "time"]]
df_thin.head()

method_id   rr      time
0  single_elastic_bm25_boost_0.25  0.0  0.110425
1  single_elastic_bm25_boost_0.25  0.0  0.071166
2  single_elastic_bm25_boost_0.25  0.0  0.051668
3  single_elastic_bm25_boost_0.25  0.0  0.048177
4  single_elastic_bm25_boost_0.25  0.0  0.049556

In [4]:
df_overall =  df_thin.groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False)
print(df_overall.to_string())

                                                                            method_id        rr      time
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  0.680024  0.174688
157                       tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.5  0.678376  0.181747
145                       tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5  0.678245  0.177712
106   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.5  0.677917  0.179724
97    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5  0.677313  0.180777
112   triple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.2_x_elastic_bm25_0.1_boost_0.5  0.675335  0.178431
163                       tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.5  0.674687  0.178412
139                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5  0.674600  0.179510
85    triple_fusion_qdrant_dense_0.4_x_qdrant_

### Overall only tuples

In [5]:
df_overall_tuple_only = df_overall[df_overall["method_id"].str.contains("tuple")]
print(df_overall_tuple_only.to_string())

                                                        method_id        rr      time
151   tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  0.680024  0.174688
157   tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.5  0.678376  0.181747
145   tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5  0.678245  0.177712
163   tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.5  0.674687  0.178412
139   tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5  0.674600  0.179510
133   tuple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.7_boost_0.5  0.668969  0.180224
169   tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.5  0.667925  0.178591
127   tuple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.8_boost_0.5  0.665149  0.180699
158  tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.75  0.662284  0.181575
164  tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.75  0.660648  0.176951
166    tuple_fusion_qdrant_dense_0.9_x_elastic_bm25_0.

### Overall only triples

In [6]:
df_overall_triple_only = df_overall[df_overall["method_id"].str.contains("triple")]
print(df_overall_triple_only.to_string())

                                                                            method_id        rr      time
106   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.5  0.677917  0.179724
97    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5  0.677313  0.180777
112   triple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.2_x_elastic_bm25_0.1_boost_0.5  0.675335  0.178431
85    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.5  0.673242  0.179853
103   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.2_x_elastic_bm25_0.2_boost_0.5  0.671694  0.179477
94    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.3_x_elastic_bm25_0.2_boost_0.5  0.670959  0.180092
115   triple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.1_x_elastic_bm25_0.1_boost_0.5  0.670185  0.180243
82    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.4_x_elastic_bm25_0.2_boost_0.5  0.668385  0.180227
70    triple_fusion_qdrant_dense_0.3_x_qdrant_

# Overall per category



In [7]:
df_thin_per_category = df[["method_id", "class_category", "rr", "time"]]
df_thin_per_category

method_id  \
0                          single_elastic_bm25_boost_0.25   
1                          single_elastic_bm25_boost_0.25   
2                          single_elastic_bm25_boost_0.25   
3                          single_elastic_bm25_boost_0.25   
4                          single_elastic_bm25_boost_0.25   
...                                                   ...   
298579  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298580  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298581  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298582  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298583  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   

                     class_category        rr      time  
0                instances_100_1000  0.000000  0.110425  
1                instances_100_1000  0.000000  0.071166  
2                instances_100_1000  0.000000  0.051668  
3                instances_100_1000  0.000000  0.048177  
4                instances_100_1000  0.000000  0.049556  
...                             ...       ...       ...  
298579  instances_100000_1000000000  1.000000  0.091356  
298580  instances_100000_1000000000  1.000000  0.087534  
298581  instances_100000_1000000000  1.000000  0.090200  
298582  instances_100000_1000000000  0.333333  0.096017  
298583  instances_100000_1000000000  1.000000  0.089052  

[298584 rows x 4 columns]

In [8]:
df_overall_per_category = df_thin_per_category.groupby(["method_id", "class_category"], as_index=False).mean().sort_values(by=["class_category","rr"], ascending=False)
print(df_overall_per_category.to_string())

                                                                             method_id               class_category        rr      time
1167                       tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5           instances_100_1000  0.654990  0.179703
1119                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5           instances_100_1000  0.654200  0.181818
1215                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5           instances_100_1000  0.652068  0.176955
783    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5           instances_100_1000  0.651780  0.184530
63                                                      single_qdrant_sparse_boost_0.5           instances_100_1000  0.651250  0.091563
975                        tuple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.9_boost_0.5           instances_100_1000  0.651244  0.183400
687    triple_fusion_qdrant_dense_0.4_x_qdrant_s

### Only instance categories

In [9]:
print(df_overall_per_category[df_overall_per_category["class_category"].str.startswith("instances")].to_string())

                                                                             method_id               class_category        rr      time
1167                       tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5           instances_100_1000  0.654990  0.179703
1119                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5           instances_100_1000  0.654200  0.181818
1215                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5           instances_100_1000  0.652068  0.176955
783    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5           instances_100_1000  0.651780  0.184530
63                                                      single_qdrant_sparse_boost_0.5           instances_100_1000  0.651250  0.091563
975                        tuple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.9_boost_0.5           instances_100_1000  0.651244  0.183400
687    triple_fusion_qdrant_dense_0.4_x_qdrant_s

### Only ancestors cetegories

In [10]:
print(df_overall_per_category[df_overall_per_category["class_category"].str.startswith("ancestors")].to_string())

                                                                             method_id            class_category        rr      time
25                                                      single_qdrant_dense_boost_0.25  ancestors_180_1000000000  0.812500  0.187582
33                                                       single_qdrant_dense_boost_0.5  ancestors_180_1000000000  0.812500  0.182235
1345                      tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.25  ancestors_180_1000000000  0.812500  0.186769
1353                       tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.5  ancestors_180_1000000000  0.812500  0.188564
1297                      tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.25  ancestors_180_1000000000  0.809295  0.187146
1305                       tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.5  ancestors_180_1000000000  0.809295  0.185529
913   triple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.1_x_elastic_bm

### Only children

In [11]:
print(df_overall_per_category[df_overall_per_category["class_category"].str.startswith("children")].to_string())

                                                                             method_id              class_category        rr      time
1211                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5       children_10000_100000  0.653882  0.178834
683    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.5       children_10000_100000  0.653256  0.181306
563    triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.6_x_elastic_bm25_0.1_boost_0.5       children_10000_100000  0.652858  0.180869
923    triple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.1_x_elastic_bm25_0.1_boost_0.5       children_10000_100000  0.651299  0.184114
659    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.4_x_elastic_bm25_0.2_boost_0.5       children_10000_100000  0.649357  0.183094
1115                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5       children_10000_100000  0.649317  0.183791
1019                       tuple_fusion_qdrant_dense_0.

# Overall - only mechanic queries

In [12]:
df_only_mech_queries = df[df["query_id"].str.startswith("mech")]
df_only_mech_queries

method_id  user_id  class_id  \
672                        single_elastic_bm25_boost_0.25        0  20825628   
673                        single_elastic_bm25_boost_0.25        0  20825628   
674                        single_elastic_bm25_boost_0.25        0  20825628   
675                        single_elastic_bm25_boost_0.25      101  20825628   
676                        single_elastic_bm25_boost_0.25      101  20825628   
...                                                   ...      ...       ...   
298579  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298580  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298581  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298582  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298583  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   

                       query_id               class_category        rr  \
672                  mech_label           instances_100_1000  1.000000   
673                   mech_desc           instances_100_1000  0.000000   
674             mech_label_desc           instances_100_1000  0.500000   
675     mech_label_user_query_0           instances_100_1000  0.055556   
676     mech_label_user_query_1           instances_100_1000  0.000000   
...                         ...                          ...       ...   
298579  mech_label_user_query_1  instances_100000_1000000000  1.000000   
298580  mech_label_user_query_2  instances_100000_1000000000  1.000000   
298581  mech_label_user_query_3  instances_100000_1000000000  1.000000   
298582  mech_label_user_query_4  instances_100000_1000000000  0.333333   
298583  mech_label_user_query_5  instances_100000_1000000000  1.000000   

            time  
672     0.050921  
673     0.118487  
674     0.149827  
675     0.120693  
676     0.097450  
...          ...  
298579  0.091356  
298580  0.087534  
298581  0.090200  
298582  0.096017  
298583  0.089052  

[165528 rows x 7 columns]

In [13]:
print(df_only_mech_queries[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())


                                                                            method_id        rr      time
139                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5  0.926017  0.188597
106   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.5  0.925982  0.188223
145                       tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5  0.925172  0.186426
97    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5  0.924912  0.190064
85    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.5  0.924826  0.189123
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  0.924436  0.181888
157                       tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.5  0.922380  0.190916
82    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.4_x_elastic_bm25_0.2_boost_0.5  0.921427  0.189997
70    triple_fusion_qdrant_dense_0.3_x_qdrant_

### Only labels

In [14]:
df_only_mech_queries_labels_only = df_only_mech_queries[df_only_mech_queries["query_id"].isin(["mech_label"])]
df_only_mech_queries_labels_only

method_id  user_id  class_id  \
672                        single_elastic_bm25_boost_0.25        0  20825628   
681                        single_elastic_bm25_boost_0.25        0   2066754   
690                        single_elastic_bm25_boost_0.25        0     34038   
699                        single_elastic_bm25_boost_0.25        0   7725634   
708                        single_elastic_bm25_boost_0.25        0    602884   
...                                                   ...      ...       ...   
298491  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0   8016240   
298500  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     23442   
298509  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     30022   
298518  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0    130019   
298527  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0  28942331   

          query_id               class_category   rr      time  
672     mech_label           instances_100_1000  1.0  0.050921  
681     mech_label         instances_1000_10000  1.0  0.068638  
690     mech_label       instances_10000_100000  1.0  0.048768  
699     mech_label  instances_100000_1000000000  1.0  0.051894  
708     mech_label        children_10000_100000  1.0  0.075789  
...            ...                          ...  ...       ...  
298491  mech_label           instances_100_1000  1.0  0.058667  
298500  mech_label  instances_100000_1000000000  1.0  0.054054  
298509  mech_label         instances_1000_10000  1.0  0.070175  
298518  mech_label       instances_10000_100000  1.0  0.064400  
298527  mech_label           instances_100_1000  1.0  0.072800  

[11088 rows x 7 columns]

In [15]:
print(df_only_mech_queries_labels_only[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr      time
157                       tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.5  0.991071  0.139908
106   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.5  0.991071  0.134191
28    triple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.7_x_elastic_bm25_0.2_boost_0.5  0.991071  0.138054
30   triple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.8_x_elastic_bm25_0.1_boost_0.25  0.991071  0.137312
127                       tuple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.8_boost_0.5  0.991071  0.138777
126                      tuple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.8_boost_0.25  0.991071  0.136078
163                       tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.5  0.991071  0.133427
70    triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.6_x_elastic_bm25_0.1_boost_0.5  0.991071  0.134484
120                      tuple_fusion_qdrant_d

### Only descriptions

In [16]:
df_only_mech_queries_desc_only = df_only_mech_queries[df_only_mech_queries["query_id"].isin(["mech_desc"])]
df_only_mech_queries_desc_only

method_id  user_id  class_id  \
673                        single_elastic_bm25_boost_0.25        0  20825628   
682                        single_elastic_bm25_boost_0.25        0   2066754   
691                        single_elastic_bm25_boost_0.25        0     34038   
700                        single_elastic_bm25_boost_0.25        0   7725634   
709                        single_elastic_bm25_boost_0.25        0    602884   
...                                                   ...      ...       ...   
298492  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0   8016240   
298501  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     23442   
298510  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     30022   
298519  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0    130019   
298528  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0  28942331   

         query_id               class_category        rr      time  
673     mech_desc           instances_100_1000  0.000000  0.118487  
682     mech_desc         instances_1000_10000  0.333333  0.101544  
691     mech_desc       instances_10000_100000  0.000000  0.088817  
700     mech_desc  instances_100000_1000000000  0.250000  0.054690  
709     mech_desc        children_10000_100000  0.000000  0.114338  
...           ...                          ...       ...       ...  
298492  mech_desc           instances_100_1000  1.000000  0.153863  
298501  mech_desc  instances_100000_1000000000  1.000000  0.113632  
298510  mech_desc         instances_1000_10000  1.000000  0.080833  
298519  mech_desc       instances_10000_100000  1.000000  0.141264  
298528  mech_desc           instances_100_1000  0.500000  0.103708  

[10692 rows x 7 columns]

In [17]:
print(df_only_mech_queries_desc_only[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr      time
193                       tuple_fusion_qdrant_sparse_0.8_x_elastic_bm25_0.2_boost_0.5  0.936728  0.115032
7                                                      single_qdrant_sparse_boost_0.5  0.936728  0.092976
121                       tuple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.9_boost_0.5  0.936728  0.193606
31    triple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.8_x_elastic_bm25_0.1_boost_0.5  0.936728  0.193186
52    triple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.7_x_elastic_bm25_0.1_boost_0.5  0.936728  0.188685
127                       tuple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.8_boost_0.5  0.936728  0.193480
196                       tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_0.1_boost_0.5  0.936728  0.108888
28    triple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.7_x_elastic_bm25_0.2_boost_0.5  0.933642  0.191793
133                       tuple_fusion_qdrant_

### Only labels with descriptions

In [18]:
df_only_mech_queries_label_desc_only = df_only_mech_queries[df_only_mech_queries["query_id"].isin(["mech_label_desc"])]
df_only_mech_queries_label_desc_only

method_id  user_id  class_id  \
674                        single_elastic_bm25_boost_0.25        0  20825628   
683                        single_elastic_bm25_boost_0.25        0   2066754   
692                        single_elastic_bm25_boost_0.25        0     34038   
701                        single_elastic_bm25_boost_0.25        0   7725634   
710                        single_elastic_bm25_boost_0.25        0    602884   
...                                                   ...      ...       ...   
298493  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0   8016240   
298502  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     23442   
298511  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0     30022   
298520  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0    130019   
298529  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...        0  28942331   

               query_id               class_category        rr      time  
674     mech_label_desc           instances_100_1000  0.500000  0.149827  
683     mech_label_desc         instances_1000_10000  0.500000  0.124570  
692     mech_label_desc       instances_10000_100000  0.142857  0.114039  
701     mech_label_desc  instances_100000_1000000000  1.000000  0.094730  
710     mech_label_desc        children_10000_100000  0.166667  0.164923  
...                 ...                          ...       ...       ...  
298493  mech_label_desc           instances_100_1000  1.000000  0.162009  
298502  mech_label_desc  instances_100000_1000000000  1.000000  0.140050  
298511  mech_label_desc         instances_1000_10000  1.000000  0.095538  
298520  mech_label_desc       instances_10000_100000  1.000000  0.148974  
298529  mech_label_desc           instances_100_1000  1.000000  0.126803  

[10692 rows x 7 columns]

In [19]:
print(df_only_mech_queries_label_desc_only[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr      time
99   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.1_x_elastic_bm25_0.3_boost_0.25  1.000000  0.195920
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  1.000000  0.191013
88    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.1_x_elastic_bm25_0.4_boost_0.5  1.000000  0.199189
85    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.5  1.000000  0.197528
84   triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.25  1.000000  0.201944
82    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.4_x_elastic_bm25_0.2_boost_0.5  1.000000  0.201561
81   triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.4_x_elastic_bm25_0.2_boost_0.25  1.000000  0.201614
148                        tuple_fusion_qdrant_dense_0.6_x_elastic_bm25_0.4_boost_0.5  1.000000  0.202201
79    triple_fusion_qdrant_dense_0.4_x_qdrant_

### Only labels with user queries

In [20]:
df_only_mech_queries_label_user_only = df_only_mech_queries[~df_only_mech_queries["query_id"].isin(["mech_desc", "mech_label", "mech_label_desc"])]
df_only_mech_queries_label_user_only

method_id  user_id  class_id  \
675                        single_elastic_bm25_boost_0.25      101  20825628   
676                        single_elastic_bm25_boost_0.25      101  20825628   
677                        single_elastic_bm25_boost_0.25      101  20825628   
678                        single_elastic_bm25_boost_0.25      101  20825628   
679                        single_elastic_bm25_boost_0.25      101  20825628   
...                                                   ...      ...       ...   
298579  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298580  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298581  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298582  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   
298583  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...      114     23442   

                       query_id               class_category        rr  \
675     mech_label_user_query_0           instances_100_1000  0.055556   
676     mech_label_user_query_1           instances_100_1000  0.000000   
677     mech_label_user_query_2           instances_100_1000  0.055556   
678     mech_label_user_query_3           instances_100_1000  0.000000   
679     mech_label_user_query_4           instances_100_1000  0.200000   
...                         ...                          ...       ...   
298579  mech_label_user_query_1  instances_100000_1000000000  1.000000   
298580  mech_label_user_query_2  instances_100000_1000000000  1.000000   
298581  mech_label_user_query_3  instances_100000_1000000000  1.000000   
298582  mech_label_user_query_4  instances_100000_1000000000  0.333333   
298583  mech_label_user_query_5  instances_100000_1000000000  1.000000   

            time  
675     0.120693  
676     0.097450  
677     0.081438  
678     0.081413  
679     0.076663  
...          ...  
298579  0.091356  
298580  0.087534  
298581  0.090200  
298582  0.096017  
298583  0.089052  

[133056 rows x 7 columns]

In [21]:
print(df_only_mech_queries_label_user_only[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr      time
106   triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.5  0.915185  0.191735
139                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5  0.914534  0.191919
145                       tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.5  0.913979  0.189245
97    triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.5  0.913655  0.193553
85    triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.5  0.913549  0.192847
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  0.913063  0.185124
112   triple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.2_x_elastic_bm25_0.1_boost_0.5  0.911580  0.191820
157                       tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.5  0.911522  0.194047
67    triple_fusion_qdrant_dense_0.3_x_qdrant_

### Overall per category

In [22]:
df_only_mech_queries_per_category_thin = df_only_mech_queries[["method_id", "class_category", "rr", "time"]]
df_only_mech_queries_per_category_thin

method_id  \
672                        single_elastic_bm25_boost_0.25   
673                        single_elastic_bm25_boost_0.25   
674                        single_elastic_bm25_boost_0.25   
675                        single_elastic_bm25_boost_0.25   
676                        single_elastic_bm25_boost_0.25   
...                                                   ...   
298579  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298580  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298581  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298582  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
298583  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   

                     class_category        rr      time  
672              instances_100_1000  1.000000  0.050921  
673              instances_100_1000  0.000000  0.118487  
674              instances_100_1000  0.500000  0.149827  
675              instances_100_1000  0.055556  0.120693  
676              instances_100_1000  0.000000  0.097450  
...                             ...       ...       ...  
298579  instances_100000_1000000000  1.000000  0.091356  
298580  instances_100000_1000000000  1.000000  0.087534  
298581  instances_100000_1000000000  1.000000  0.090200  
298582  instances_100000_1000000000  0.333333  0.096017  
298583  instances_100000_1000000000  1.000000  0.089052  

[165528 rows x 4 columns]

In [23]:
df_only_mech_queries_overall_per_category = df_only_mech_queries_per_category_thin.groupby(["method_id", "class_category"], as_index=False).mean().sort_values(by=["class_category", "rr"], ascending=False)
print(df_only_mech_queries_overall_per_category.to_string())

                                                                             method_id               class_category        rr      time
1575                       tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_0.1_boost_0.5           instances_100_1000  0.898201  0.117994
63                                                      single_qdrant_sparse_boost_0.5           instances_100_1000  0.895160  0.095525
975                        tuple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.9_boost_0.5           instances_100_1000  0.895001  0.194590
1119                       tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.5           instances_100_1000  0.893371  0.192812
255    triple_fusion_qdrant_dense_0.1_x_qdrant_sparse_0.8_x_elastic_bm25_0.1_boost_0.5           instances_100_1000  0.892541  0.195667
1071                       tuple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.7_boost_0.5           instances_100_1000  0.890872  0.194023
1023                       tuple_fusion_qdrant_d

# Overall - only user queries

In [24]:
df_only_user_queries = df[df["query_id"].str.startswith("user")]
df_only_user_queries

method_id  user_id  class_id  \
0                          single_elastic_bm25_boost_0.25        1  20825628   
1                          single_elastic_bm25_boost_0.25        1  20825628   
2                          single_elastic_bm25_boost_0.25        1  20825628   
3                          single_elastic_bm25_boost_0.25        1  20825628   
4                          single_elastic_bm25_boost_0.25        1  20825628   
...                                                   ...      ...       ...   
297743  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14     23442   
297744  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14     23442   
297745  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14     23442   
297746  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14     23442   
297747  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14     23442   

            query_id               class_category    rr      time  
0       user_query_0           instances_100_1000  0.00  0.110425  
1       user_query_1           instances_100_1000  0.00  0.071166  
2       user_query_2           instances_100_1000  0.00  0.051668  
3       user_query_3           instances_100_1000  0.00  0.048177  
4       user_query_4           instances_100_1000  0.00  0.049556  
...              ...                          ...   ...       ...  
297743  user_query_1  instances_100000_1000000000  1.00  0.078431  
297744  user_query_2  instances_100000_1000000000  1.00  0.069612  
297745  user_query_3  instances_100000_1000000000  0.00  0.081128  
297746  user_query_4  instances_100000_1000000000  0.00  0.088692  
297747  user_query_5  instances_100000_1000000000  0.25  0.082815  

[133056 rows x 7 columns]

In [25]:
print(df_only_user_queries[["method_id", "rr", "time"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr      time
158                      tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.75  0.444588  0.171155
164                      tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.75  0.444284  0.164017
170                      tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.75  0.443438  0.166795
5                                                      single_qdrant_dense_boost_0.75  0.437833  0.167235
152                      tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.75  0.434964  0.168799
113  triple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.2_x_elastic_bm25_0.1_boost_0.75  0.425746  0.168450
146                      tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.75  0.423870  0.165279
116  triple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.1_x_elastic_bm25_0.1_boost_0.75  0.422524  0.168946
107  triple_fusion_qdrant_dense_0.6_x_qdrant_s

### Overall per category

In [26]:
df_only_user_queries_per_category_thin = df_only_user_queries[["method_id", "class_category", "rr", "time"]]
df_only_user_queries_per_category_thin

method_id  \
0                          single_elastic_bm25_boost_0.25   
1                          single_elastic_bm25_boost_0.25   
2                          single_elastic_bm25_boost_0.25   
3                          single_elastic_bm25_boost_0.25   
4                          single_elastic_bm25_boost_0.25   
...                                                   ...   
297743  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
297744  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
297745  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
297746  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   
297747  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...   

                     class_category    rr      time  
0                instances_100_1000  0.00  0.110425  
1                instances_100_1000  0.00  0.071166  
2                instances_100_1000  0.00  0.051668  
3                instances_100_1000  0.00  0.048177  
4                instances_100_1000  0.00  0.049556  
...                             ...   ...       ...  
297743  instances_100000_1000000000  1.00  0.078431  
297744  instances_100000_1000000000  1.00  0.069612  
297745  instances_100000_1000000000  0.00  0.081128  
297746  instances_100000_1000000000  0.00  0.088692  
297747  instances_100000_1000000000  0.25  0.082815  

[133056 rows x 4 columns]

In [27]:
df_only_user_queries_overall_per_category = df_only_user_queries_per_category_thin.groupby(["method_id", "class_category"], as_index=False).mean().sort_values(by=["class_category", "rr"], ascending=False)
print(df_only_user_queries_overall_per_category.to_string())

                                                                             method_id               class_category        rr      time
1367                      tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.75           instances_100_1000  0.422273  0.167460
1271                      tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.75           instances_100_1000  0.419903  0.170317
1319                      tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.75           instances_100_1000  0.419316  0.163747
47                                                      single_qdrant_dense_boost_0.75           instances_100_1000  0.416155  0.166111
1223                      tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.75           instances_100_1000  0.407858  0.168067
1175                      tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.75           instances_100_1000  0.396451  0.164661
1127                      tuple_fusion_qdrant_de

### Choosing the best match per class of a user

- The idea is that if user writes the 6 queries, and at least one query find the result, the user can use it to model the data structure.
- We want to choose the best query for each of the user assigned classes.
- We can do it by grouping by: method, user, and class.
  - Then we select a maximum for the groups from reciprocal rank.
  - Then the results will be used to again group by a method. 

In [28]:
df_only_user_queries_max_per_user_class = df_only_user_queries[["method_id", "user_id", "class_id", "rr"]].groupby(["method_id", "user_id", "class_id"], as_index=False).max()
df_only_user_queries_max_per_user_class

method_id  user_id  class_id  \
0                         single_elastic_bm25_boost_0.25        1     34038   
1                         single_elastic_bm25_boost_0.25        1    602884   
2                         single_elastic_bm25_boost_0.25        1   2066754   
3                         single_elastic_bm25_boost_0.25        1   4897819   
4                         single_elastic_bm25_boost_0.25        1   7725634   
...                                                  ...      ...       ...   
22171  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14   8016240   
22172  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14  11483816   
22173  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14  18593264   
22174  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14  28942331   
22175  tuple_fusion_qdrant_sparse_0.9_x_elastic_bm25_...       14  51591359   

             rr  
0      0.000000  
1      0.000000  
2      0.000000  
3      0.000000  
4      0.000000  
...         ...  
22171  0.333333  
22172  0.500000  
22173  0.500000  
22174  1.000000  
22175  1.000000  

[22176 rows x 4 columns]

### Average rr with only best ranks per classes of users


In [29]:
print(df_only_user_queries_max_per_user_class[["method_id", "rr"]].groupby(["method_id"], as_index=False).mean().sort_values(by="rr", ascending=False).to_string())

                                                                            method_id        rr
158                      tuple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.3_boost_0.75  0.792347
164                      tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.75  0.786905
152                      tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.75  0.782377
170                      tuple_fusion_qdrant_dense_0.9_x_qdrant_sparse_0.1_boost_0.75  0.779773
5                                                      single_qdrant_dense_boost_0.75  0.777466
107  triple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.3_x_elastic_bm25_0.1_boost_0.75  0.772406
113  triple_fusion_qdrant_dense_0.7_x_qdrant_sparse_0.2_x_elastic_bm25_0.1_boost_0.75  0.770685
146                      tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.75  0.769728
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  0.765795
145                       tuple_fusion_q

### Number of classes found per method with only best ranks

- In total we have 56 classes and each class is given to two users.
- Each method have a change of scoring 112 found classes.

In [30]:
df_only_user_queries_max_per_user_class_thin = df_only_user_queries_max_per_user_class[["method_id", "rr"]].copy()
# Round up rr meaning the method found the class.
df_only_user_queries_max_per_user_class_thin["rr"] = df_only_user_queries_max_per_user_class_thin["rr"].apply(lambda x: math.ceil(x))
# Group the results by method -> for a method, each class is presented twice (since two users provided the queries).
print(df_only_user_queries_max_per_user_class_thin.groupby(["method_id"], as_index=False).sum().sort_values(by="rr", ascending=False).to_string())


                                                                            method_id   rr
98   triple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.4_x_elastic_bm25_0.1_boost_0.75  104
164                      tuple_fusion_qdrant_dense_0.8_x_qdrant_sparse_0.2_boost_0.75  104
70    triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.6_x_elastic_bm25_0.1_boost_0.5  104
46    triple_fusion_qdrant_dense_0.2_x_qdrant_sparse_0.5_x_elastic_bm25_0.3_boost_0.5  104
68   triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.5_x_elastic_bm25_0.2_boost_0.75  104
146                      tuple_fusion_qdrant_dense_0.5_x_qdrant_sparse_0.5_boost_0.75  104
65   triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.4_x_elastic_bm25_0.3_boost_0.75  104
64    triple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.4_x_elastic_bm25_0.3_boost_0.5  104
133                       tuple_fusion_qdrant_dense_0.3_x_qdrant_sparse_0.7_boost_0.5  104
151                       tuple_fusion_qdrant_dense_0.6_x_qdrant_sparse_0.4_boost_0.5  104

### Classes the best methods did not find (selection from no rerank analysis)

- triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1
- tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6

In [31]:
# Triple fusion triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1

df1_triple_best = df_only_user_queries_max_per_user_class[df_only_user_queries_max_per_user_class["method_id"].str.contains("triple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.5_x_elastic_bm25_0.1_boost_0.75")][["method_id", "class_id", "rr"]].copy()
df1_triple_best["rr"] = df_only_user_queries_max_per_user_class_thin["rr"].apply(lambda x: math.ceil(x))
df1_triple_best[df1_triple_best["rr"] == 0]

method_id   class_id  rr
9635  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...    4897819   0
9636  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...    7725634   0
9637  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...   18557321   0
9647  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...  123691918   0
9651  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...    4897819   0
9661  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...   20755546   0
9663  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...  123691918   0
9701  triple_fusion_qdrant_dense_0.4_x_qdrant_sparse...    1195098   0

- The class it did not find at all:
    - 4897819 - role
    - 123691918 - tool

In [32]:
# Tuple fusion tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6

df1_triple_best = df_only_user_queries_max_per_user_class[df_only_user_queries_max_per_user_class["method_id"].str.contains("tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_0.6_boost_0.75")][["method_id", "class_id", "rr"]].copy()
df1_triple_best["rr"] = df_only_user_queries_max_per_user_class_thin["rr"].apply(lambda x: math.ceil(x))
df1_triple_best[df1_triple_best["rr"] == 0].drop_duplicates()

method_id   class_id  rr
15683  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...    4897819   0
15684  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...    7725634   0
15685  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...   18557321   0
15695  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...  123691918   0
15709  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...   20755546   0
15749  tuple_fusion_qdrant_dense_0.4_x_qdrant_sparse_...    1195098   0

- The classes it did not find at all
    - 4897819 - role
    - 123691918 - tool

Interestingly, they are the same as before. Looking into the queries, we tried them manually, and it did not find them.
Thinking, maybe it was since the things were too abstract, the users have a big problem with generating the queries to.

# The least found classes

In [33]:
df[["class_id", "class_category", "rr"]].groupby(["class_id", "class_category"], as_index=False).mean().sort_values(by="rr", ascending=False)

class_id               class_category        rr
12     130019       instances_10000_100000  0.928882
51   51591359       instances_10000_100000  0.811794
48   29654788  instances_100000_1000000000  0.802609
29    3007842         instances_1000_10000  0.761753
46   23002039         instances_1000_10000  0.761584
39   15052846           instances_100_1000  0.751415
28    2819161           instances_100_1000  0.736428
45   21191270  instances_100000_1000000000  0.732026
22    1348305       instances_10000_100000  0.718077
24    1371150           instances_100_1000  0.714530
4       30022         instances_1000_10000  0.710107
1       11303         instances_1000_10000  0.709003
36   11483816         instances_1000_10000  0.708157
27    2319498       instances_10000_100000  0.704505
25    1478437         instances_1000_10000  0.691947
38   13442814  instances_100000_1000000000  0.689685
5       34038       instances_10000_100000  0.676968
20    1134686       instances_10000_100000  0.673236
50   35922124           instances_100_1000  0.658156
30    3231690       instances_10000_100000  0.639537
17     310899         instances_1000_10000  0.638968
16     294414        children_10000_100000  0.635836
19    1076052         instances_1000_10000  0.633221
41   18593264  instances_100000_1000000000  0.626240
3       24856         instances_1000_10000  0.610727
47   28942331           instances_100_1000  0.606499
23    1364302           instances_100_1000  0.604300
0         532  instances_100000_1000000000  0.603893
33    5783996       instances_10000_100000  0.602124
49   30325611            ancestors_150_180  0.597720
26    2066754         instances_1000_10000  0.574633
2       23442  instances_100000_1000000000  0.557109
42   19389637  instances_100000_1000000000  0.553119
6       34763       instances_10000_100000  0.552958
15     179049       instances_10000_100000  0.548001
52   63125083           instances_100_1000  0.539089
9       47521  instances_100000_1000000000  0.530556
43   20755546     ancestors_180_1000000000  0.528595
7       43229  instances_100000_1000000000  0.515621
10      79007  instances_100000_1000000000  0.513854
37   12890393           instances_100_1000  0.509970
40   18557321            ancestors_150_180  0.509917
55  124030970     ancestors_180_1000000000  0.505406
14     134556  instances_100000_1000000000  0.494911
31    3621673         instances_1000_10000  0.489053
34    7725634  instances_100000_1000000000  0.475632
44   20825628           instances_100_1000  0.469361
18     602884        children_10000_100000  0.410029
21    1195098       instances_10000_100000  0.362873
8       44539         instances_1000_10000  0.348250
53  123523193           instances_100_1000  0.324363
11     124714       instances_10000_100000  0.311720
35    8016240           instances_100_1000  0.298889
13     133036           instances_100_1000  0.244093
32    4897819   children_100000_1000000000  0.219320
54  123691918   children_100000_1000000000  0.175050